apps/wallet/src/ui/app/redux/slices/txresults/index.ts

```diff
@@ -69,16 +70,27 @@ const deduplicate = (results: string[] | undefined) =>
         ? results.filter((value, index, self) => self.indexOf(value) === index)
         : [];
 
-// TODO: This is a temporary solution to get the NFT data from Call txn
-const getCreatedObjectID = (txEffects: TransactionEffects): string | null => {
-    return txEffects?.created
-        ? txEffects?.created.map((item) => item.reference)[0]?.objectId
-        : null;
-};
-
 const moveCallTxnName = (moveCallFunctionName?: string): string | null =>
     moveCallFunctionName ? moveCallFunctionName.replace(/_/g, ' ') : null;
 
+// Get objectId from a transaction effects -> events where recipient is the address
+const getTxnEffectsEventID = (
+    txEffects: TransactionEffects,
+    address: string
+): string[] => {
+    const events = txEffects?.events || [];
+    const objectIDs = events
```

Comment:
Do we really want to go through every event group?

Reply:
Do you mean stop the loop when we find the first matching item?

Reply:
I mean more is everything in this `events` object useful to look at?